compiler/rustc_const_eval/src/interpret/place.rs

```diff
@@ -1013,30 +1013,50 @@ where
         )
     }
 
-    /// Allocates a string in the interpreter's memory, returning it as a (wide) place.
-    /// This is allocated in immutable global memory and deduplicated.
-    pub fn allocate_str_dedup(
-        &mut self,
-        str: &str,
-    ) -> InterpResult<'tcx, MPlaceTy<'tcx, M::Provenance>> {
-        let bytes = str.as_bytes();
-        let ptr = self.allocate_bytes_dedup(bytes)?;
-
+    /// Combines pointer and metadata into a wide pointer.
+    fn slice_from_ptr_and_len(
+        &self,
+        ptr: Pointer<M::Provenance>,
+        len: usize,
+        slice_ty: Ty<'tcx>,
+    ) -> MPlaceTy<'tcx, M::Provenance> {
```

Comment:
I'm not a fan of having more single-user functions in this core file.

Reply:
Removed.